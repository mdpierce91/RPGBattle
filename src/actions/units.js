import database from '../firebase/firebase';
import { getRandomUnits } from '../models/UnitRandomizer';

export const createRandomUnit = () => {
    console.log('create a new unit!');
    let unit = getRandomUnits(1)[0];
    console.log('new unit:', unit);
    return addUnit(unit)
}

export const createRandomUnits = () => {
    console.log('create a new units!');
    let units = getRandomUnits(25);
    console.log('new units:', units);
    return addUnits(units);
}

export const addUnit = (unit) => {
    console.log('dispatch add unit:', unit);
    return {
        type: 'ADD_UNIT',
        unit
    }
};

export const addUnits = (units) => ({
    type: 'ADD_UNITS',
    units
});

export const startSetUnits = (unitSet) => {
    console.log('get data from firebase');
    return (dispatch) => {
        database.ref(`units/${unitSet}`).once('value').then((data) => {    
            console.log('data fetched from firebase', data);
            dispatch({
                type: 'POPULATE_UNITS',
                team: data
            })
        });
    }
}