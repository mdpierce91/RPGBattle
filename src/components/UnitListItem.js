import React from 'react';

const UnitListItem = ({...unit}) => {
    return (
        <div>
            <span>{unit.name}</span>
            <span>{unit.id}</span>
        </div>
    );
}

export { UnitListItem as default };