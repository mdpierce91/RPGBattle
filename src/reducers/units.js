const unitsReducer = (state=[], action) => {
    switch (action.type){
        case 'POPULATE_UNITS':    
            return action.units;
        case 'ADD_UNIT':
            console.log('add unit: ', action.unit);
            return [
                ...state,
                action.unit
            ];
        case 'ADD_UNITS':
            console.log('add units: ', action.units);
            return [
                ...state,
                ...action.units
            ];
        case 'REMOVE_UNIT':
            return state.filter((unit) => unit.id === action.id);
        default:
            return state;
    }
}

export { unitsReducer as default };