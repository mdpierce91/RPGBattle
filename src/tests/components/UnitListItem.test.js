import React from 'react';
import { shallow } from 'enzyme';
import UnitListItem from '../../components/UnitListItem';
import units from '../fixtures/units';

let wrapper;

test('should render unit list item', () => {
    wrapper = shallow(<UnitListItem unit={units[0]}/>);
    expect(wrapper).toMatchSnapshot();
})