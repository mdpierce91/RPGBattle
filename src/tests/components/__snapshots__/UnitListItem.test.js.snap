// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render unit list item 1`] = `
<div>
  <span />
  <span />
</div>
`;
